src/js/editor/editor.js

```diff
@@ -70,6 +70,16 @@ const CALLBACK_QUEUES = {
   INPUT_MODE_DID_CHANGE: 'inputModeDidChange'
 };
 
+// There are only two events that we're concerned about for Undo, that is inserting text and deleting content.
+// These are the only two states that go on a "run" and create a combined undo, everything else has it's own
+// deadicated undo.
+const EDITOR_RUN_LAST_ACTION_STATE = {
+  INSERT_TEXT: 1,
+  DELETE: 2
+};
+
+const UNDO_RUN_BREAK_TIMEOUT_LENGTH = 500; // how many ms after the last event before an UNDO run is broken
```

Comment:
Mentioned in gitter: Increasing this seems worthwhile. 5000 ms is a good default. I'm wondering if it makes sense to allow it to be customizable, also...